state/factory/workingset.go

```diff
@@ -591,17 +595,24 @@ func (ws *workingSet) pickAndRunActions(
 				actionIterator.PopAccount()
 				continue
 			}
-			// TODO: error may cause the tx to be stuck in the action pool until expired
 			receipt, err := ws.runAction(actionCtx, nextAction)
 			switch errors.Cause(err) {
 			case nil:
 				// do nothing
-			case action.ErrChainID:
+			case action.ErrGasLimit:
+				actionIterator.PopAccount()
```

Comment:
In an extreme scenario, if this is a huge action that exceeds the entire block gas limit, then this transaction will remain in the actpool, blocking this user's transaction indefinitely.